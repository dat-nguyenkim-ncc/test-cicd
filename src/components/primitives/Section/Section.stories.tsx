import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Section, { SectionProps } from './Section'
import { Box } from 'theme-ui'

export default {
  title: 'Primitives/Section',
  component: Section,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<SectionProps> = args => (
  <Box sx={{ height: '100vh', p: 5, width: '100%', backgroundColor: 'gray03' }}>
    <Section {...args} />
  </Box>
)

export const Default = Template.bind({})
Default.args = {
  children: 'Section',
} as SectionProps
