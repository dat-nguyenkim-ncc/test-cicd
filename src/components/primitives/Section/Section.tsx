import React from 'react'
import { Box } from 'theme-ui'
import { ViewInterface } from '../../../types'

export type SectionProps = ViewInterface<{}>

const Section = ({ children, sx }: SectionProps) => {
  return (
    <Box
      sx={{
        padding: 6,
        backgroundColor: 'background',
        borderRadius: 10,
        maxWidth: 1024,
        ...sx,
      }}
    >
      {children}
    </Box>
  )
}

export default Section
