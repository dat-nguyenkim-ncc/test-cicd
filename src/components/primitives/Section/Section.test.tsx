import * as React from 'react'
import { shallow } from 'enzyme'
import Section from './Section'

describe('Section', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Section />)
    expect(wrapper).toMatchSnapshot()
  })
})
