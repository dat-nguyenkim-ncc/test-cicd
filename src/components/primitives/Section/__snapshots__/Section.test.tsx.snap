// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches snapshot 1`] = `ShallowWrapper {}`;
